tensorflow/lite/micro/kernels/cmsis_nn/add.cc

```diff
@@ -302,8 +341,10 @@ TfLiteStatus EvalAdd(TfLiteContext* context, TfLiteNode* node) {
   TFLITE_DCHECK(node->user_data != nullptr);
   const OpData* data = static_cast<const OpData*>(node->user_data);
 
-  if (output->type == kTfLiteFloat32) {
-    EvalAddFloat(context, node, params, data, input1, input2, output);
+  if (output->type == kTfLiteFloat32 ||
+      (input1->type == kTfLiteInt32 && input2->type == kTfLiteInt32)) {
```

Comment:
Is it sufficient/equivalent to just check `output->type`? It's a bit more readable as it's what the rest of the function does.

Reply:
I can switch to use `output->type` for consistency.

Longer term we may need a larger refactor where we want to check the quantization params. For example, you could have a quantized add and unquantized add that both emit `int16`.